src/control/common/cmdutil/logging.go

```diff
@@ -24,9 +28,83 @@ type (
 	LogCmd struct {
 		logging.Logger
 	}
+
+	// LogConfig contains parameters used to configure the logger.
+	LogConfig struct {
+		LogFile  string
+		LogLevel common.ControlLogLevel
+		JSON     bool
+	}
 )
 
 // SetLog sets the logger for the command.
 func (cmd *LogCmd) SetLog(log logging.Logger) {
 	cmd.Logger = log
 }
+
+// ConfigureLogger configures the logger according to the requested config.
+func ConfigureLogger(logIn logging.Logger, cfg LogConfig) error {
+	log, ok := logIn.(*logging.LeveledLogger)
+	if !ok {
+		return errors.New("logger is not a LeveledLogger")
+	}
+
+	// Set log level mask for default logger from config,
+	// unless it was explicitly set to debug via CLI flag.
+	applyLogConfig := func() error {
+		switch logging.LogLevel(cfg.LogLevel) {
+		case logging.LogLevelTrace:
+			log.SetLevel(logging.LogLevelTrace)
+			log.Debugf("Switching control log level to TRACE")
+		case logging.LogLevelDebug:
+			log.SetLevel(logging.LogLevelDebug)
+			log.Debugf("Switching control log level to DEBUG")
+		case logging.LogLevelNotice:
+			log.Debugf("Switching control log level to NOTICE")
+			log.SetLevel(logging.LogLevelNotice)
+		case logging.LogLevelError:
+			log.Debugf("Switching control log level to ERROR")
+			log.SetLevel(logging.LogLevelError)
+		}
+
+		if cfg.JSON {
+			log = log.WithJSONOutput()
+		}
+
+		log.Debugf("configured logging: level=%s, file=%s, json=%v",
+			cfg.LogLevel, cfg.LogFile, cfg.JSON)
+
+		return nil
+	}
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return errors.Wrap(err, "getting hostname")
+	}
+
+	// Set log file for default logger if specified in config.
+	if cfg.LogFile != "" {
+		f, err := common.AppendFile(cfg.LogFile)
+		if err != nil {
+			return errors.Wrap(err, "create log file")
+		}
+
+		log.Infof("%s logging to file %s",
```

Comment:
nit: unnecessary line break

Reply:
I'll fix it if I need to re-push. Thanks.